gadopt/energy_solver.py

```diff
@@ -68,10 +70,22 @@ def __init__(self, T, u, approximation,
                     weak_bc[type] = value
             self.weak_bcs[id] = weak_bc
 
-        self.ts = timestepper(self.eq, T, self.fields, delta_t, self.weak_bcs, strong_bcs=self.strong_bcs,
-                              solver_parameters=self.solver_parameters)
-        self.T_old = self.ts.solution_old
+        self.timestepper = timestepper
+        self.T_old = Function(self.Q)
+        # solver is setup only last minute
+        # so people can overwrite parameters we've setup here
+        self._solver_setup = False
+
+    def setup_solver(self):
+        """Setup timestepper and associated solver"""
+        self.ts = self.timestepper(self.eq, self.T, self.fields, self.delta_t,
+                                   bnd_conditions=self.weak_bcs, solution_old=self.T_old,
+                                   strong_bcs=self.strong_bcs,
+                                   solver_parameters=self.solver_parameters)
+        self._solver_setup = True
```

Comment:
I guess an alternative would be to just check for the presence of `self.ts`, but this is probably better for enforcing construction of the timestepper.